tests/foreman/api/test_docker.py

```diff
@@ -1303,16 +1308,18 @@ def test_positive_create_with_name(self):
         """
         for name in valid_data_list():
             with self.subTest(name):
-                url = gen_url(subdomain=gen_string('alpha'))
                 description = gen_string('alphanumeric')
                 registry = entities.Registry(
                     description=description,
                     name=name,
-                    url=url,
+                    url=self.url,
                 ).create()
-                self.assertEqual(registry.name, name)
-                self.assertEqual(registry.url, url)
-                self.assertEqual(registry.description, description)
+                try:
+                    self.assertEqual(registry.name, name)
+                    self.assertEqual(registry.url, self.url)
+                    self.assertEqual(registry.description, description)
+                finally:
+                    registry.delete()
```

Comment:
``self.subTest`` is a context manager and we can wrap code in a chain of context managers, so we can create a new one e.g: **CleanUp**.  

```
@contextlib.contextmanager
def CleanUp(self):
    try:
        yield
    finally:
        # execute the addCleanup's here
```
so

```python
with self.subTest(name), self.CleanUp():
     Registry.create()
     self.assertEqual(registry.name, name)
     ...
``` 

The only advantage I see is the standard use of teardowns, I think that kind of solution will only be needed when we have a lot of cases like this, for this isolated case I guess it is worthless.